src/middlewared/middlewared/etc_files/local/sysrrd_disable.py

```diff
@@ -0,0 +1,28 @@
+import os
+import tarfile
+
+
+SYSRRD_SENTINEL = '/data/sentinels/sysdataset-rrd-disable'
+
+
+def rename_tarinfo(tarinfo):
+    name = tarinfo.name.split('/', maxsplit=4)
+    tarinfo.name = f'collectd/rrd/{"" if len(name) < 5 else name[-1]}'
+    return tarinfo
+
+
+def sysrrd_disable(middleware):
```

Comment:
Seems like this script should only run after system dataset has started?

Reply:
Also this does not seem like a good fit more a etc plugin, perhaps its makes more sense as a private method in system dataset plugin.